source/postprocess/visualization/maximum_shear_stress.cc

```diff
@@ -0,0 +1,280 @@
+            Assert(eigenvector_array[1].first > eigenvector_array[2].first, ExcMessage("the eigenvalue in array 1 should be larger then in array 2."));
+
+            // Check whether the vectors are orthogonal
+            Assert(std::fabs(eigenvector_array[0].second * eigenvector_array[1].second) < 1e-8, ExcMessage("Eigenvector in 0 is not orthogonal to eigenvector 1:"
+                   "eigenvector_array[1].second * eigenvector_array[1].second = " + std::to_string(eigenvector_array[0].second * eigenvector_array[1].second)));
+            Assert(std::fabs(eigenvector_array[1].second * eigenvector_array[2].second) < 1e-8, ExcMessage("Eigenvector in 1 is not orthogonal to eigenvector 2:"
+                   "eigenvector_array[1].second * eigenvector_array[2].second = " + std::to_string(eigenvector_array[1].second * eigenvector_array[2].second)));
+            Assert(std::fabs(eigenvector_array[0].second * eigenvector_array[2].second) < 1e-8, ExcMessage("Eigenvector in 0 is not orthogonal to eigenvector 2:"
+                   "eigenvector_array[0].second * eigenvector_array[2].second = " + std::to_string(eigenvector_array[0].second * eigenvector_array[2].second)));
+
+            const Tensor<1,dim> tau_direction = eigenvector_array[2].second - eigenvector_array[0].second;
+
+            const double tau_magnitude = 0.5 * (eigenvector_array[0].first - eigenvector_array[2].first);
+
+            const Tensor<1,dim> tau = (tau_direction/tau_direction.norm()) * tau_magnitude;
+
+
+            for (unsigned int i=0; i<dim; ++i)
+              computed_quantities[q](i) = tau[i];
+          }
+      }
+
+
+      template <int dim>
+      std::vector<std::string>
+      MaximumShearStress<dim>::get_names () const
+      {
+        return std::vector<std::string> (dim, "maximum_shear_stress");
+      }
+
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      MaximumShearStress<dim>::get_data_component_interpretation () const
+      {
+        return
+          std::vector<DataComponentInterpretation::DataComponentInterpretation>
+          (dim,
+           DataComponentInterpretation::component_is_part_of_vector);
+      }
+
+
+
+      template <int dim>
+      UpdateFlags
+      MaximumShearStress<dim>::get_needed_update_flags () const
+      {
+        return update_gradients | update_values | update_q_points;
+      }
+
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(MaximumShearStress,
+                                                  "maximum shear stress",
+                                                  "A plugin that computes the direction and magnitude of the "
+                                                  "maximum horizontal component of the compressive stress as a vector "
```

Comment:
I think you'll have to read through this again and see which parts need updating.